wallet/wallet-cli-lib/src/commands/mod.rs

```diff
@@ -231,382 +246,501 @@ fn print_coin_amount(chain_config: &ChainConfig, value: Amount) -> String {
+                        controller.encrypt_wallet(&None).map_err(WalletCliError::Controller)?;
+                    }
+                }
 
-        WalletCommand::UnlockPrivateKeys { password } => {
-            match controller_opt.as_mut() {
-                None => {
-                    return Err(WalletCliError::NoWallet);
+                Ok(ConsoleCommand::Print("Success".to_owned()))
+            }
+
+            WalletCommand::UnlockPrivateKeys { password } => {
+                match controller_opt.as_mut() {
+                    None => {
+                        return Err(WalletCliError::NoWallet);
+                    }
+                    Some(controller) => {
+                        controller.unlock_wallet(&password).map_err(WalletCliError::Controller)?;
+                    }
                 }
-                Some(controller) => {
-                    controller.unlock_wallet(&password).map_err(WalletCliError::Controller)?;
+
+                Ok(ConsoleCommand::Print("Success".to_owned()))
+            }
+
+            WalletCommand::LockPrivateKeys => {
+                match controller_opt.as_mut() {
+                    None => {
+                        return Err(WalletCliError::NoWallet);
+                    }
+                    Some(controller) => {
+                        controller.lock_wallet().map_err(WalletCliError::Controller)?;
+                    }
                 }
+
+                Ok(ConsoleCommand::Print("Success".to_owned()))
             }
 
-            Ok(ConsoleCommand::Print("Success".to_owned()))
-        }
+            WalletCommand::ChainstateInfo => {
+                let info = rpc_client.chainstate_info().await.map_err(WalletCliError::RpcError)?;
+                Ok(ConsoleCommand::Print(format!("{info:#?}")))
+            }
 
-        WalletCommand::LockPrivateKeys => {
-            match controller_opt.as_mut() {
-                None => {
-                    return Err(WalletCliError::NoWallet);
-                }
-                Some(controller) => {
-                    controller.lock_wallet().map_err(WalletCliError::Controller)?;
-                }
+            WalletCommand::BestBlock => {
+                let id = rpc_client.get_best_block_id().await.map_err(WalletCliError::RpcError)?;
+                Ok(ConsoleCommand::Print(id.hex_encode()))
             }
 
-            Ok(ConsoleCommand::Print("Success".to_owned()))
-        }
+            WalletCommand::BestBlockHeight => {
+                let height =
+                    rpc_client.get_best_block_height().await.map_err(WalletCliError::RpcError)?;
+                Ok(ConsoleCommand::Print(height.to_string()))
+            }
 
-        WalletCommand::ChainstateInfo => {
-            let info = rpc_client.chainstate_info().await.map_err(WalletCliError::RpcError)?;
-            Ok(ConsoleCommand::Print(format!("{info:#?}")))
-        }
+            WalletCommand::BlockHash { height } => {
```

Comment:
I know this was the same as before your change, but should this command be changed to "BlockId" rather than "BlockHash"?

Reply:
Is there a reason why BlockHash instead of Id is used?

Reply:
No particular reason, I just used the name from Bitcoin Core. @OBorce please update it.